tox.ini

```diff
@@ -4,12 +4,67 @@
 # and then run "tox" from this directory.
 
 [tox]
-envlist = py37, py38, py39, py310
+skip_missing_interpreters = true
+envlist =
+    lint
+    python-azdev-unit
 
-[testenv]
+[base]
 deps =
+    -r dev_requirements
+description =
+# py{x} determines python version for environment.
+#    A valid interpreter must be installed (and added to PATH)
+#    for each selected version, otherwise the run will be skipped
+# az{x} determines:
+#    {min} - minimum supported azure-cli
+#    {cur} - latest released azure-cli
+#    {dev} - installs azure-cli from ../azure-cli
+# int/unit determines test suites to run
+# list all available tox environments with: `tox -av`
+    python: Local Python
+    py3.7: Python 3.7
+    py3.8: Python 3.8
+    py3.9: Python 3.9
+    py3.10: Python 3.10
+    azmin: min azure-cli
+    azcur: current azure-cli
+    azdev: dev azure-cli
+    int: Integration Tests
+    unit: Unit Tests
+
+[testenv:lint]
+description = run linter
+deps =
+    {[base]deps}
     azure-cli
 commands =
-    pip install -r dev_requirements
-    pip install --upgrade ../azure-cli/src/azure-cli-testsdk
-    pytest azext_iot/tests
+    flake8 azext_iot/
+    pylint azext_iot/ --rcfile=.pylintrc
+
+[testenv:py{thon,3.7,3.8,3.9,3.10}-az{min,cur,dev}-{int,unit}]
+skip_install = True
+passenv =
+    AZEXT_IOT_TESTRG
+description =
+    {[base]description}
+deps =
+    # base deps
+    {[base]deps}
+    # azure-cli deps
+    azmin: azure-cli==2.32.0
+    azcur: azure-cli
+    azdev: ../azure-cli/src/azure-cli
+    azdev: ../azure-cli/src/azure-cli-core
+    # azure cli test sdk
+    ../azure-cli/src/azure-cli-testsdk
+commands =
+    python --version
+    # install to tox extension dir
+    pip install -U --target {envsitepackagesdir}/azure-cli-extensions/azure-iot .
+    # validate az and extension version
+    az -v
+    # run tests
+    # You can pass additional positional args to pytest using `-- {args}`
+    unit: pytest -k _unit ./azext_iot/tests {posargs}
+    int: pytest -k _int ./azext_iot/tests {posargs}
```

Comment:
have we thought about leveraging python-xist as we do today in existing pipelines to parallelize worker processes ?

Reply:
Looks like `tox` has some parallelization built-in  https://tox.wiki/en/latest/user_guide.html#parallel-mode

Reply:
I haven't gotten too far into integration test parallelization yet, as I'm thinking we may want to plan out some more `testenv` separation / distinction for individual or groups of tests (similar to how we've moved tests into aspect folders).

This PR mostly captures the pipeline work for unit tests (with the capability to run int tests locally as part of tox as well).
In order to run tox for integration tests locally, I also believe we need to add our sizeable environment variable list into the tox environment `passenv` section as well. More to come on this in future iterations.

I've had mixed results with Tox's parallelization running unit tests - for a few reasons, but namely that debug output isn't printed during runs (which doesn't hurt as much locally as it does in a pipeline) and also some weird inconsistencies that could be due to how i'm setting up the separate test environments.